resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -255,13 +255,17 @@ private[spark] object KubernetesConf {
     s"spark-${UUID.randomUUID().toString.replaceAll("-", "")}"
 
   def getResourceNamePrefix(appName: String): String = {
-    val id = KubernetesUtils.uniqueID()
-    s"$appName-$id"
-      .trim
-      .toLowerCase(Locale.ROOT)
-      .replaceAll("[^a-z0-9\\-]", "-")
-      .replaceAll("-+", "-")
-      .replaceAll("^-", "")
+    var prefix = ""
+    while (prefix.isEmpty) {
```

Comment:
Can we avoid `while` because sometime this may lead us to the infinite loop if someone change the body of `while` accidentally in the future?

Reply:
Thank you, @dongjoon-hyun. 

Add `while` here is because the current body may produce empty string `""`. To avoid while loop, we need to change the algorithm:

- how do you think @yaooqinn‘s suggestion, simply replaces the illegal characters with 'x'
- or if the result is `""`, just return a constant string, e.g. `"spark"`

Reply:
option 1 is adopted